docs/coding-guidelines/javascript.md

```diff
@@ -125,54 +123,105 @@ Indentation and line breaks add readability to complex statements.
 
 Tabs should be used for indentation.
 
+Try to return early from a function to avoid deeply tabbed functions akin to ["Callback Hell"](http://callbackhell.com/).
+
+```js
+// bad
+function getSiteTitle( site ) {
+	if ( site ) {
+		if ( site.name ) {
+			if ( site.name.length <= 50 ) {
+				return site.name;
+			} else {
+				return site.name.slice( 0, 50 ) + '…';
+			}
+		} else {
+			return site.slug;
+		}
+	} else {
+		return null;
+	}
+}
+
+
+// good
+function getSiteTitle( site ) {
+	if ( ! site ) {
+		return null;
+	}
+
+	if ( ! site.name ) {
+		return site.slug;
+	}
+
+	if ( site.name.length > 50 ) {
+		return site.name.slice( 0, 50 ) + '…';
+	}
+
+	return site.name;
```

Comment:
maybe there's another example we could use. I feel like here this encourages big code due to the null-checking when we could collapse it all down. maybe some function checking options on a passed argument?

```js
if ( ! data.isLoading ) {
	return;
}

if ( ! data.hasLoaded ) {
	return;
}

…
```

also, there could be a bug in here because we don't truncate the slug…